examples/multi-threading.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: UTF-8 -*-
+
+from fbchat import Client
+from fbchat.models import *
+import threading
+
+# Subclass fbchat.Client and override required methods
+class PrintMessage(Client):
+    def onMessage(self, author_id, message_object, thread_id, thread_type, **kwargs):
+        self.markAsDelivered(thread_id, message_object.uid)
+        self.markAsRead(thread_id)
+
+        print("\nIncoming message: {}".format(message_object.text))
+
+    # Creating some feedback on login since we will disable INFO logging
+    def onLoggedIn(self, email=None):
+        print("Login of {} successful.".format(email))
+
+
+# Logging in and setting logging level to WARNING to avoid some unessential output
+client = PrintMessage("<email>", "<password>", logging_level=30)
+
+
+def send():
+    while True:
+        payload = input("Message: ")
+        if payload:
+            client.send(
```

Comment:
Again, I don't think operating the listener and sending messages on the same client is thread safe! At least not formally. So I'm not sure if it's something we wanna publicly recommend?

Reply:
I assume you mean it's not safe to send a message while listening? So by stopping the listening thread while sending, we should be OK?

Reply:
That would probably be enough! But on the other hand, it's really inefficient, and not really clean either 😕. Maybe the best solution is just to create two clients?